### PR TITLE
Make RenderContext itself a Sink.

### LINKED ISSUE #812: Simplify Kotlin WorkflowAction
This is a simpler alternative to #644.

`WorkflowAction` is nice conceptually, but its Kotlin incarnation still requires too much ceremony. A handful of fixes should go a long way to streamlining its use:

- [ ] Eliminate `EventSink`, and instead make `RenderContext` itself the sink. (Or at least make `RenderContext` implement the interface itself, s.t. we can preserve `contramap` and such.)
- [x] Instead of returning the output value, give `Mutator` an output field that defaults to null. (Take care to disallow setting it to null.)
- [x] Give `StatelessWorkflow` the same workflowAction extension that `StatefulWorkflow` has
- [x] ~Provide a ktlint check that warns when event handler lambdas do anything other than call `context.update`~ (moved to #834)

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Closes #812.

### CODE REVIEW

File: kotlin/workflow-core/src/main/java/com/squareup/workflow/RenderContext.kt [hunk 1]
```diff
@@ -26,6 +26,7 @@ import com.squareup.workflow.WorkflowAction.Updater
  *
  * ## Handling Events
  *
+ * TODO update
```
Comment (zach-klippenstein) [inline kotlin/workflow-core/src/main/java/com/squareup/workflow/RenderContext.kt]: TODO @zach-klippenstein Update docs

File: kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt [hunk 2]
```diff
@@ -97,15 +97,13 @@ class TodoEditorWorkflow : StatelessWorkflow<TodoList, TodoEditorOutput, TodoRen
...
-    val rawSink = context.makeActionSink<TodoAction>()
-
     // Make event handling idempotent until https://github.com/square/workflow/issues/541 is fixed.
     var eventFired = false
     val sink = object : Sink<TodoAction> {
       override fun send(value: TodoAction) {
         if (eventFired) return
         eventFired = true
-        rawSink.send(value)
+        context.send(value)
```
Comment (rjrjr) [inline kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt]: I'm increasingly filled with regret that we didn't go with `RenderContext.render(…)` as the method signature.

Comment (rjrjr) [reply kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt]: Just thinking out loud, don't want to distract us from this release, but I bet we could retrofit that pretty easily. Same trick we're doing with `WorkflowAction` — introduce the new method with a default impl that calls the old method, etc.

File: kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/PlayerWorkflow.kt [hunk 3]
```diff
@@ -65,15 +65,11 @@ class PlayerWorkflow(
...
-    return Rendering(
-        actorRendering = ActorRendering(avatar = avatar, movement = state),
-        onStartMoving = { sink.send(StartMoving(it)) },
-        onStopMoving = { sink.send(StopMoving(it)) }
-    )
-  }
+  ): Rendering = Rendering(
+      actorRendering = ActorRendering(avatar = avatar, movement = state),
+      onStartMoving = { context.send(StartMoving(it)) },
+      onStopMoving = { context.send(StopMoving(it)) }
```
Comment (kirillzh) [inline kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/PlayerWorkflow.kt]: 😍

File: kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt (hunk 2 above)
Comment (zach-klippenstein) [reply kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt]: For sure. See #836.

File: kotlin/workflow-core/src/main/java/com/squareup/workflow/RenderContext.kt (hunk 1 above)
Comment (zach-klippenstein) [reply kotlin/workflow-core/src/main/java/com/squareup/workflow/RenderContext.kt]: Done.

**Comment by zach-klippenstein** [comment]
Addressed @rjrjr's comments, and also cleaned up `RealRenderContextTest`'s actions that were using the deprecated `apply` method.

**Pull request merged**